### PR TITLE
GPS CNAV message decoding support.

### PR DESCRIPTION
**Pull request by valeri-atamaniouk**
Added capability to decode GPS CNAV message and extract the following information: CNAV message header, symbol delay.

### CODE REVIEW

File: include/libswiftnav/cnav_msg.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+} cnav_msg_decoder_t;
+
+void cnav_msg_decoder_init(cnav_msg_decoder_t *dec);
+bool cnav_msg_decoder_add_symbol(cnav_msg_decoder_t *dec,
+                                 unsigned char symbol,
+                                 cnav_msg_t *msg,
+                                 u32 *delay);
+
+// test methods
+void _cnav_bitlshift(void *buf, size_t bits, u32 shift);
```
Comment (ljbade) [inline include/libswiftnav/cnav_msg.h]: @fnoble Do we normally expose functions for testing in the header file?

**Comment by ljbade** [comment]
Apart from my query about exposing the test functions, this PR looks good to me.

**Comment by rge-exafore** [comment]
1. Commit message doesn't match the format requirements:
- period in the end of summary line
- Write the summary line and description of what you have done in the imperative mode

2. SBAS-related constants in cnav_mgs.c are unnecessary

Otherwise looks good to me.

**Comment by valeri-atamaniouk** [comment]
Will refactor

**Pull request closed without merging**